internal/primitives/database/database.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package database
+
+import "github.com/ChainSafe/gossamer/internal/primitives/runtime"
+
+// ColumnID is an identifier for a column.
+type ColumnID uint32
+
+// Change is an alteration to the database.
+type Change[H any] any
+
+// ChangeTypes is the interface constraint which can be a Change
+type ChangeTypes[H any] interface {
+	ChangeSet | ChangeRemove | ChangeStore[H] | ChangeReference[H] | ChangeRelease[H]
```

Comment:
nit/could the "change" be removed from these so they are just "Set", "Remove", etc

Reply:
It's not clear that this is a `Change` type.  On second thought I may use a superfluous interface to represent `Change` given this doesn't need to be encoded or decoded.